native_client/deepspeech.cc

```diff
@@ -304,23 +304,38 @@ DS_FreeModel(ModelState* ctx)
 }
 
 int
-DS_EnableDecoderWithLM(ModelState* aCtx,
-                       const char* aLMPath,
-                       const char* aTriePath,
-                       float aLMAlpha,
-                       float aLMBeta)
+DS_EnableExternalScorer(ModelState* aCtx,
+                        const char* aScorerPath)
 {
   aCtx->scorer_.reset(new Scorer());
-  int err = aCtx->scorer_->init(aLMAlpha, aLMBeta,
-                                aLMPath ? aLMPath : "",
-                                aTriePath ? aTriePath : "",
-                                aCtx->alphabet_);
+  int err = aCtx->scorer_->init(aScorerPath, aCtx->alphabet_);
   if (err != 0) {
     return DS_ERR_INVALID_LM;
   }
   return DS_ERR_OK;
 }
 
+int
+DS_DisableExternalScorer(ModelState* aCtx)
```

Comment:
I'm still unsure why we would need that API entry point. If the user does not call `DS_EnableExternalScorer`, that's enough, right ? Do we have feedback from people switching it on and off on the same context ?

Reply:
My thought was that applications might want to release the memory allocated by the external scorer without having to destroy the model entirely. And it's simple enough to implement. If you think it doesn't make sense to expose, I'm happy to remove it.

Reply:
Let's keep it for now then, it makes sense.

Reply:
Yes, disabling then enabling a different scorer works and lets you switch scorers. The only catch is that scorer instances are attached to the Model, not the Stream, and are re-used by all streams. So switching scorers mid-stream could lead to all kinds of weird behavior. I need to check this more closely.

Reply:
I checked and this is indeed a bit more complicated than how it's currently handled by this PR. The way I see it, we have two options: require that every stream using the Scorer has finished before allowing one to disable it, or making it so that the actual scorer instance is only destroyed when the last stream using it is finished.

The former is an awkward API, as it's pretty much a non-recoverable failure. The latter can lead to memory leaks if an application doesn't close its streams. I think an application not closing its streams is not a common scenario, as all of the "Finish" functions take care of that. So I'm partial to just deferring scorer destruction until the last stream is finished. 

@lissyx @kdavis-mozilla any opinions?

Reply:
The latter seems to make much more sense to me. Leaks will happen when people do shit anyway, as long as we don't generate them that's fine. The former looks like a recipe to shoot into one's foot with a RPG.

Reply:
+1 on the latter